packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -48,19 +52,46 @@ export async function createNewFluidFile(
     logger: ITelemetryLogger,
     createNewSummary: ISummaryTree | undefined,
     epochTracker: EpochTracker,
+    fileEntry: IFileEntry,
+    createNewCaching: boolean,
 ): Promise<IOdspResolvedUrl> {
     // Check for valid filename before the request to create file is actually made.
     if (isInvalidFileName(newFileInfo.filename)) {
         throwOdspNetworkError("Invalid filename. Please try again.", invalidFileNameStatusCode);
     }
 
-    const itemId = createNewSummary === undefined
-        ? await createNewEmptyFluidFile(getStorageToken, newFileInfo, logger, epochTracker)
-        : await createNewFluidFileFromSummary(getStorageToken, newFileInfo, logger, createNewSummary, epochTracker);
+    let itemId: string;
+    let id: string = "";
+
+    if (createNewSummary === undefined) {
+        itemId = await createNewEmptyFluidFile(getStorageToken, newFileInfo, logger, epochTracker);
+    } else {
+        const content = await createNewFluidFileFromSummary(
+            getStorageToken, newFileInfo, logger, createNewSummary, epochTracker);
+        itemId = content.itemId;
+        id = content.id;
+    }
 
     const odspUrl = createOdspUrl({... newFileInfo, itemId, dataStorePath: "/"});
     const resolver = new OdspDriverUrlResolver();
-    return resolver.resolve({ url: odspUrl });
+    const odspResolvedUrl = await resolver.resolve({ url: odspUrl });
+    fileEntry.docId = odspResolvedUrl.hashedDocumentId;
+    fileEntry.resolvedUrl = odspResolvedUrl;
+
+    if (createNewSummary !== undefined && createNewCaching) {
+        assert(id !== undefined, "response content id is undefined");
```

Comment:
Can you change the message to "Summary handle is undefined" and also rename id to summaryHandle.